netket/vqs/mc/mc_state/expect_forces.py

```diff
@@ -75,9 +75,9 @@ def forces_expect_hermitian(
     local_value_args: PyTree,
 ) -> Tuple[PyTree, PyTree]:
 
-    σ_shape = σ.shape
-    if jnp.ndim(σ) != 2:
-        σ = σ.reshape((-1, σ_shape[-1]))
+    n_chains = σ.shape[0]
+    if σ.ndim >= 3:
+        σ = jax.lax.collapse(σ, 0, 2)
```

Comment:
?

Reply:
Here I only merge the first two axes, and not more (which would be convenient if we ever decide to have non flat samples). However to make it equal to before we would have to change it to `jax.lax.collapse(σ, 0, σ.ndim-1)`.

I thought using `jax.lax.collapse` was cleaner, and the reshape emmitted at the end is the same.

(I will do something similar in SR in the next PR, where I have to jit the reshape, for which jax.lax.collapse is convenient.)